source/Ship.cpp

```diff
@@ -2055,9 +2055,9 @@ void Ship::DoGeneration()
 	// uninterrupted scanning is required to successfully scan its target.
 	// Only apply the decay if not already done scanning the target.
 	if(cargoScan < SCAN_TIME)
-		cargoScan = max(0., cargoScan - 1.);
+		cargoScan = max(0., cargoScan * 0.999 - 1.);
 	if(outfitScan < SCAN_TIME)
-		outfitScan = max(0., outfitScan - 1.);
+		outfitScan = max(0., outfitScan * 0.999 - 1.);
```

Comment:
Playing around with the idea of low-power scanners simply being unable to scan very large ships. Also, it makes the tail end of the scan take somewhat longer than the start, which one could describe as "climactic", but it's not _very_ climactic.

Perhaps a better implementation would be to test how long it would take to scan (assuming no interference via movement) and immediately return "no, this scan's not gonna work" if it's below a certain threshold (in this case, if the increase to `elapsed` is less than `SCAN_TIME * (1. - 0.999)`), and to return to linear decay.